### PR TITLE
Chore: Unite wire graphs for cli and server

### PR DESCRIPTION
**Pull request by zserge**
Since both CLI and the server has been united into the same binary, there is little sense in keeping two separate Wire graphs for them. Or, to be more precise - a common set of dependencies for both CLI and Server should be singular.

This PR removes the CLI runner wire set and shares that with the server base wire set.

As part of this PR I had to move runner.Runner into server.Runner to avoid cyclic dependencies. Maybe it would make more sense to rename `pkg/server` into `pkg/wire` and put Server and Runner into `pkg/wire/server.Server` and `pkg/wire/runner.Runner` at some point to resolve this.

### CODE REVIEW

File: pkg/cmd/grafana-cli/commands/commands.go [hunk 1]
```diff
@@ -10,30 +10,34 @@ import (
...
 		if err != nil {
-			return fmt.Errorf("%v: %w", "failed to load configuration", err)
+			return fmt.Errorf("%v: %w", "failed to initialize runner", err)
 		}
 
-		r, err := runner.Initialize(cfg)
-		if err != nil {
-			return fmt.Errorf("%v: %w", "failed to initialize runner", err)
+		if cmd.Bool("debug") {
+			runner.Cfg.LogConfigSources()
```
Comment (DanCech) [inline pkg/cmd/grafana-cli/commands/commands.go:37]: Any particular reason to inline this in 2 places?

**Comment by DanCech** [comment]
I'd definitely support moving wire out of server as you described, lgtm otherwise!

File: pkg/cmd/grafana-cli/commands/commands.go (hunk 1 above)
Comment (zserge) [reply pkg/cmd/grafana-cli/commands/commands.go:37]: Previously it's been a shared function, now since the order of injection has changed (instead of Cfg we take CommandLineArgs, and use Cfg from the instantiated Runner) - I decided to have this code duplicated. But yeah, there is a way to unify this again.

**Comment by zserge** [comment]
> I'd definitely support moving wire out of server as you described, lgtm otherwise!

I'd rather have it in a separate PR, as it would change imports in quite a few files, so I'd keep wire set change and import rename change as two separate changesets.

File: pkg/server/wire.go [hunk 2]
```diff
@@ -379,6 +379,20 @@ var wireSet = wire.NewSet(
 	wire.Bind(new(oauthtoken.OAuthTokenService), new(*oauthtoken.Service)),
 )
 
+var wireCLISet = wire.NewSet(
```
Comment (mildwonkey) [inline pkg/server/wire.go:382]: this might be a good time to add some comments about the difference between these sets, and suggestions for what new things go where (if you have an idea. I don't).

Comment (zserge) [reply pkg/server/wire.go:382]: Ideally, everything should go into `wireBasicSet`. CLISet I hope would not be needed soon (need to look into enterprise) and it could be instantiated from the same BasicSet graph as would the Server.
Now, wireTestSet is a question. I'd prefer to see it gone and let unit tests inject dependencies (mock) manually, as most of them do already. Integration tests on the other hand should rather use real implementations when possible. We should be able to move prefs and sqlstore into the basic set. NGAlert and Notifications are terra incognita to me, but I hope there would be a solution to mock them in a similar manner without a separate wire graph.
TLDR: I hope there would be one wireSet once this is all finished. What we have now is there "for legacy reasons".

Comment (suntala) [reply pkg/server/wire.go:379]: I'm still not sure the extent to which we interact with these sets on a day-to-day basis, but I second documenting these guidelines somehow. Your comment seems like a good overview already. The documentation could be something along the lines of the [Queries with Result Fields](https://github.com/grafana/grafana/blob/15804234d6ea8d803def4edf802d142e418979f1/contribute/backend/communication.md#queries-with-result-fields) section, which describes how to refactor away Result fields. I suppose the guidelines you describe could be added to the [Services](https://github.com/grafana/grafana/blob/15804234d6ea8d803def4edf802d142e418979f1/contribute/backend/services.md) section, which already includes an introduction to wire.

**Review by papagian — APPROVED** [review]
